### PR TITLE
feat(e2ei): add networking layer for acme server

### PR DESCRIPTION
**Pull request by mchenani**
----
#### PR Submission Checklist for internal contributors

- The **PR Title**
  - [x] conforms to the style of semantic commits messages¹ supported in Wire's Github Workflow²
  - [ ] contains a reference JIRA issue number like `SQPIT-764`
  - [x] answers the question: _If merged, this PR will: ..._ ³

- The **PR Description**
  - [ ] is free of optional paragraphs and you have filled the relevant parts to the best of your ability
----

# What's new in this PR?
* Networking layer for ACME server
* Add ignoreSSLCert option to httpClient [this is only enabled for development staging]
### Issues

The current ACME server is a test server without a valid SSL cert.
----
#### PR Post Submission Checklist for internal contributors (Optional)

 - [ ] Wire's Github Workflow has automatically linked the PR to a JIRA issue
----
#### PR Post Merge Checklist for internal contributors

 - [ ] If any soft of configuration variable was introduced by this PR, it has been added to the relevant documents and the CI jobs have been updated.
----
##### References
1. https://sparkbox.com/foundry/semantic_commit_messages
1. https://github.com/wireapp/.github#usage
1. E.g. `feat(conversation-list): Sort conversations by most emojis in the title #SQPIT-764`.

### CODE REVIEW

File: network/src/commonJvmAndroid/kotlin/com/wire/kalium/network/HttpEngine.kt [hunk 1]
```diff
@@ -28,15 +28,27 @@ import java.net.Authenticator
...
-    proxyCredentials: ProxyCredentialsDTO?
+    proxyCredentials: ProxyCredentialsDTO?,
+    ignoreAllSSLErrors: Boolean
 ): HttpClientEngine = OkHttp.create {
     // OkHttp doesn't support configuring ping intervals dynamically,
     // so they must be set when creating the Engine
     // See https://youtrack.jetbrains.com/issue/KTOR-4752
+
+    val clientBuilder =
+        OkHttpClient.Builder().ignoreAllSSLErrors().pingInterval(WEBSOCKET_PING_INTERVAL_MILLIS, TimeUnit.MILLISECONDS)
```
[the remark this answers is not in the document]
Comment (mchenani) [reply network/src/commonJvmAndroid/kotlin/com/wire/kalium/network/HttpEngine.kt:48]: Oh, thanks. fixing it.

File: network/src/commonMain/kotlin/com/wire/kalium/network/api/base/authenticated/e2ei/E2EIApi.kt [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+        newAccountRequestBody: List<UByte>
+    ): NetworkResponse<String>
+
+    suspend fun sendNewAuthz(): NetworkResponse<Unit>
+
+    suspend fun sendNewOrder(): NetworkResponse<Unit>
+
+    suspend fun sendAuthzHandle(): NetworkResponse<Unit>
+
+    suspend fun sendAuthzClienId(): NetworkResponse<Unit>
```
Comment (typfel) [inline network/src/commonMain/kotlin/com/wire/kalium/network/api/base/authenticated/e2ei/E2EIApi.kt:39]:
```suggestion
    suspend fun sendAuthzClientId(): NetworkResponse<Unit>
```

File: network/src/commonMain/kotlin/com/wire/kalium/network/api/base/authenticated/e2ei/E2EIApi.kt [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+
+    companion object {
+        fun getApiNotSupportError(apiName: String, apiVersion: String = "4") = NetworkResponse.Error(
```
Comment (typfel) [inline network/src/commonMain/kotlin/com/wire/kalium/network/api/base/authenticated/e2ei/E2EIApi.kt:42]: Nice, don't have to solve it now but would it possible to share this function with all API interfaces? Move it to a base API interface?

File: logic/src/commonMain/kotlin/com/wire/kalium/logic/featureFlags/KaliumConfigs.kt [hunk 4]
```diff
@@ -37,6 +37,7 @@ data class KaliumConfigs(
     val blacklistHost: String = "",
     val maxAccount: Int = 0,
     val developmentApiEnabled: Boolean = false,
+    val ignoreAllSSLErrors: Boolean = false,
```
Comment (typfel) [inline logic/src/commonMain/kotlin/com/wire/kalium/logic/featureFlags/KaliumConfigs.kt:40]: So will this be removed once we have more proper development setup?

**Review by typfel — CHANGES_REQUESTED** [review]

**Pull request closed without merging**